contribution_evaluation.py

```
@@ -0,0 +1,304 @@
+                end = function.end_line
+                c = change['changed']
+                start_pos = bisect.bisect_left(c, start)
+                if start_pos < len(c) and c[start_pos] <= end:
+                    complexity_data.append({
+                        "function_name": function.name,
+                        "complexity": function.cyclomatic_complexity,
+                        "lines": function.length,
+                        'start_line': function.start_line,
+                        'end_line': function.end_line,
+                    })
+                    print(f"  ——The information of changed function——")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)